Armadillo/src/main/java/com/scribd/armadillo/download/DownloadEngine.kt

```diff
@@ -70,15 +70,20 @@ internal class ExoplayerDownloadEngine @Inject constructor(
         scope.launch(errorHandler) {
             launch {
                 // Download DRM license for offline use
-                offlineDrmManager.downloadDrmLicenseForOffline(audioPlayable.request)
+                offlineDrmManager.downloadDrmLicenseForOffline(id = audioPlayable.id.toString(), request = audioPlayable.request)
             }
 
             launch {
-                val downloadHelper = downloadHelper(context, audioPlayable.request)
+                val downloadHelper = downloadHelper(
+                    id = audioPlayable.id.toString(),
+                    context = context,
+                    mediaRequest = audioPlayable.request
+                )
 
                 downloadHelper.prepare(object : DownloadHelper.Callback {
                     override fun onPrepared(helper: DownloadHelper) {
-                        val request = helper.getDownloadRequest(audioPlayable.id.encodeInByteArray())
+                        var request = helper.getDownloadRequest(audioPlayable.id.encodeInByteArray())
+                        request = request.copyWithId(audioPlayable.id.toString())
```

Comment:
Why are we changing from the byte array to the string?

Reply:
It's string everywhere else, and the `audioplayable.id` is *not* what's being set as the ID there, its being set as the URL, and then i have to override it back to being the `audioplayable.id` instead.

Reply:
Out of scope: I don't think we use the `audioPlayable.id.encodeInByteArray()` data anywhere, it's a bit of a red herring. I think we can pass `null`

Also, the `var` threw me off for a moment thinking there was reason to reassign it.

Any reason not to do:

`val request = helper.getDownloadRequest(audioPlayable.id.encodeInByteArray()).copyWithId(audioPlayable.id.toString())`?